extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BeanArchiveProcessor.java

```diff
@@ -149,11 +149,12 @@ private IndexView buildApplicationIndex(ArcConfig config, ApplicationArchivesBui
                         .map(bda -> new BeanDefiningAnnotation(bda.getName(), bda.getDefaultScope()))
                         .collect(Collectors.toList()), stereotypes);
         beanDefiningAnnotations.addAll(customScopes.getCustomScopeNames());
-        // Also include archives that are not bean archives but contain scopes, qualifiers or interceptor bindings
+        // Also include archives that are not bean archives but contain scopes, qualifiers, interceptor bindings or decorators
         beanDefiningAnnotations.add(DotNames.SCOPE);
         beanDefiningAnnotations.add(DotNames.NORMAL_SCOPE);
         beanDefiningAnnotations.add(DotNames.QUALIFIER);
         beanDefiningAnnotations.add(DotNames.INTERCEPTOR_BINDING);
+        beanDefiningAnnotations.add(DotNames.DECORATOR);
```

Comment:
I think that we should add the `@Interceptor` as well. It could happen that an archive contains an interceptor class but not an interceptor binding definition. Note that this particular set is not used as "regular" bean defining annotations, i.e. meta-annotations are not handled specifically...

Reply:
Right, I was thinking about that too. We already check for bindings, you'd need to have an archive with just interceptor and a binding coming from yet another archive? But sure, there is no reason not to add it.

Reply:
> you'd need to have an archive with just interceptor and a binding coming from yet another archive?

Yes, that's  exactly the use case I'm talking about...

Reply:
Agree we should add `@Interceptor` too. Maybe also `@Stereotype`?